pkg/http/server.go

```diff
@@ -0,0 +1,68 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package http
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/http/pprof"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/submariner-io/admiral/pkg/log"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var logger = log.Logger{Logger: logf.Log}
+
+type StopFunc func()
+
+// StartMetricsAndProfileServer starts an HTTP server providing access to Prometheus metrics
+// and/or profiling information.
+// The returned function should be called to shut down the server, typically at process exit
+// using defer in the main() function.
+func StartMetricsAndProfileServer(metrics, profile bool, port int) StopFunc {
```

Comment:
nit - "MetricsAndProfile" are options (passed as params) and I don't think need/should to be part of the name. Also technically it should be "MetricsAndOrProfile" 😄  

```suggestion
func StartServer(metrics, profile bool, port int) StopFunc {
```

Reply:
I actually didn’t like the use of bools, so let’s avoid those altogether.